web/content/docs/benchmarks/thermo-hydro-mechanics/Thermohydromechanical_Process.md

```diff
@@ -0,0 +1,23 @@
++++
+project = "Thermohydromechanics/Linear/"
+author = "Tianyuan Zheng"
+date = "2017-08-23T20:39:58+01:00"
+title = "Thermohydromechanics"
+weight = 101
+
+[menu]
+  [menu.benchmarks]
+    parent = "Thermohydromechanics"
+
++++
+
+{{< project-link >}}
+
+## Problem description
+
+This benchmark is dedicated to simulate the thermohydromechanical process in porous media. Four different tests are implemented. For the detailed problem settings and results description, please refer to the attached document.
```

Comment:
Comments on the benchmarks:
1. Point injection: 
     This is an example of applying heat power at a point in an infinite fully saturated 3D domain. To solve the example properly by using the FEM, we have to restrict the problem in a bounded space and fix the displacement on the boundary.   It is OK to solve it with asymmetrical  assumption but the quarter circle  domain causes a a little problem. The problem is that it is difficulty  to apply normal displacement on curved boundary. That's why this example is normally solved in cubic domain, which can  easily be applied fixed normal displacement BC on the outer boundary.  In this point injection benchmark, the example is  solved as an asymmetrical  problem but  the normal displacement BC on the arc boundary is committed. With such simplification, we can only get a solutions that are roughly close to the analytical ones if the size of the analyzed domain is relative small. On the other hand, to obtain an accurate solution of this example by the FEM is impossible due to the singularity in the temperature and pressure   at the injecting point. In other words, the analytical solution of this example can only be approximated by using FEM. To improve the approximation of this benchmark, you can replace the quarter circle  domain with a square domain, which means to cut a cylinder from the infinite space under the assumption of axisymmetry. With the square domain, the fixed normal displacement BC can be easily applied onto the outer boundary. However, I will not expect a significant improvement in the solution accuracy with the changing of the domain and BC.  

2. Bi material benchmark,
With the current implementation, the fluid thermal expansion always exists if solid thermal expansion is given. That means the benchmark is solved as a transient problem in one step, which brings a slight perturbation to the linear distribution of variables along the 1D domain. Without the solid thermal expansion, the benchmark is meaning less. Therefore, suggest to add an option for fluid thermal expansion in the source code, or leave an explanation in the benchmark description.

Reply:
@grubbymoon Can you act on Wenqing's comment? We need some of these explanations in the pdf (and to actually perform/test some of the improvements). I briefly scanned through the documentation pdf: Please replace Figs. 3 and 4 by quantitative line plots (contour plots are not readable). And as usual, Figs. 6 to 8 need error measures (and the accompanying measures outlined at the beginning of my comment). Fig. 9 is very good to show but is not readable --> replot with meaningful scales.